scripts/find_localised_strings.py

```diff
@@ -0,0 +1,282 @@
+"""Search all given paths recursively for localised string calls."""
+import argparse
+import ast
+import json
+import pathlib
+import re
+import sys
+import dataclasses
+from typing import Optional
+import io
+
+
+def get_func_name(thing: ast.AST) -> str:
+    """Get the name of a function from a Call node."""
+    if isinstance(thing, ast.Name):
+        return thing.id
+
+    elif isinstance(thing, ast.Attribute):
+        return get_func_name(thing.value)
+
+    else:
+        return ""
+
+
+def get_arg(call: ast.Call) -> str:
+    """Extract the argument string to the translate function."""
+    if len(call.args) > 1:
+        print("??? > 1 args", call.args, file=sys.stderr)
+
+    arg = call.args[0]
+    if isinstance(arg, ast.Constant):
+        return arg.value
+    elif isinstance(arg, ast.Name):
+        return f"VARIABLE! CHECK CODE! {arg.id}"
+    else:
+        return f'Unknown! {type(arg)=} {ast.dump(arg)} ||| {ast.unparse(arg)}'
+
+
+def find_calls_in_stmt(statement: ast.AST) -> list[ast.Call]:
+    """Recursively find ast.Calls in a statement."""
+    out = []
+    for n in ast.iter_child_nodes(statement):
+        out.extend(find_calls_in_stmt(n))
+    if isinstance(statement, ast.Call) and get_func_name(statement.func) == "_":
+
+        out.append(statement)
+
+    return out
+
+
+COMMENT_RE = re.compile(r'^.*?(#.*)$')
+
+
+def extract_comments(call: ast.Call, lines: list[str], file: pathlib.Path) -> Optional[str]:
+    out: list[Optional[str]] = []
+    above = call.lineno - 2
+    current = call.lineno - 1
+
+    above_line = lines[above].strip() if len(lines) < above else None
+    current_line = lines[current].strip()
+
+    for line in (above_line, current_line):
+        if line is None or '#' not in line:
+            out.append(None)
+            continue
+
+        match = COMMENT_RE.match(line)
+        if not match:
+            print(line)
+            out.append(None)
+            continue
+
+        comment = match.group(1).strip()
+        if not comment.startswith("# LANG:"):
+            print(f'Unknown comment for {file}:{current} {line}', file=sys.stderr)
+            out.append(None)
+            continue
+
+        out.append(comment.replace("# LANG:", "").strip())
+
+    if out[1] is not None:
+        return out[1]
+    elif out[0] is not None:
+        return out[0]
+
+    return None
+
+
+def scan_file(path: pathlib.Path) -> list[ast.Call]:
+    """Scan a file for ast.Calls."""
+    data = path.read_text()
```

Comment:
`path.read_text(encoding='utf-8')` else it crashes out on hotkey.py in the `DISPLAY` dict.